Compiler/BackEnd/Initialization.mo

```diff
@@ -816,6 +816,9 @@ protected
   HashSet.HashSet hs;
   list<DAE.ComponentRef> crefs;
 algorithm
+  // lochel: workaround to align all elements
+  globalKnownVars := BackendVariable.listVar(BackendVariable.varList(globalKnownVars));
```

Comment:
Basically, yes. In fact, it would be nice to fix the `Variables` data structure (or the utility functions that work on it). I think it would be even better to replace the data structure completely by using the `ExpandableArray`. That way, we would have a generic implementation which means less redundant code and less maintenance effort.